test/oli_web/live/experiments_live_test.exs

```diff
@@ -91,133 +88,220 @@ defmodule OliWeb.ExperimentsLiveTest do
       redirect_path =
         "/authoring/session/new?request_path=%2Fauthoring%2Fproject%2F#{project.slug}%2Fexperiments"
 
-      {:error,
-       {:redirect,
-        %{
-          to: ^redirect_path
-        }}} =
+      {:error, {:redirect, %{to: ^redirect_path}}} =
         live(conn, live_view_experiments_route(project.slug))
     end
   end
 
   describe "experiments view" do
-    setup [:admin_conn, :create_project]
+    setup [:admin_conn, :create_project, :put_route]
 
-    test "loads experiments view correctly", %{
-      conn: conn,
-      project: project
-    } do
-      {:ok, view, _html} = live(conn, live_view_experiments_route(project.slug))
+    test "loads experiments view correctly", %{view: view} do
+      {view, %{}}
+      |> step(:test_has_title_AB_testing)
+      |> step(:test_has_message_integrate_with_AB_platform)
+      |> step(:test_has_checkbox)
+    end
+
+    test "when clicked on checkbox creates and displays experiment with 2 options", %{view: view} do
+      {view, %{}}
+      |> step(:test_has_alternatives_group, :refute)
+      |> step(:check_on_checkbox)
+      |> step(:test_has_alternatives_group)
+      |> step(:test_has_options)
+      |> step(:put_resource_id)
+      |> step(:put_options)
+      |> step(:test_has_button_show_edit_group_modal)
+      |> step(:test_has_button_show_edit_option_1_modal)
+      |> step(:test_has_button_show_edit_option_2_modal)
+      |> step(:test_has_button_show_delete_option_1_modal)
+      |> step(:test_has_button_show_delete_option_2_modal)
+      |> step(:test_has_new_option_link)
+    end
 
-      assert view
-             |> element("h3")
-             |> render() =~
-               "A/B Testing with UpGrade"
-
-      assert view
-             |> element("p")
-             |> render() =~
-               "To support A/B testing, Torus integrates with the A/B testing platform"
-
-      assert has_element?(
-               view,
-               "label",
-               "Enable A/B testing with UpGrade"
-             )
+    test "when checkbox is off then disable the ability to modify experiment", %{view: view} do
+      {view, %{}}
+      |> step(:check_on_checkbox)
+      |> step(:check_on_checkbox)
+      |> step(:test_has_button_show_edit_group_modal, :refute)
+      |> step(:test_has_button_show_edit_option_1_modal, :refute)
+      |> step(:test_has_button_show_edit_option_2_modal, :refute)
+      |> step(:test_has_button_show_delete_option_1_modal, :refute)
+      |> step(:test_has_button_show_delete_option_2_modal, :refute)
+      |> step(:test_has_new_option_link, :refute)
```

Comment:
If I understood correctly this way of piping the test with steps, I think it would be impossible to find all those elements in the view since the assertion depends on having the options and resource ids in the context map. I mean, we are trying to find some elements in the view with a selector that depends on those ids (phx-value-option-id and phx-value-resource-id)

Reply:
I think we could get the needed resource and options ids by fetching the experiment in the db, or maybe change the way we define those element selectors to not depend on their ids.

Reply:
It makes sense. We need the `resource_id` to get or not a match. Let me see how can I adapt it.

Reply:
@nicocirio It was relatively easy to fix using the piping approach... Check out this [commit](https://github.com/Simon-Initiative/oli-torus/pull/4788/commits/ee230ea1457283adab272158341b37f3b7d55dcb). I'm starting to like it.